sequencer/addrqueue.go

```diff
@@ -128,13 +129,20 @@ func (a *addrQueue) updateCurrentNonceBalance(nonce *uint64, balance *big.Int) (
 	return a.readyTx, oldReadyTx
 }
 
-// UpdateTxZKCounters updates the ZKCounters for the given tx (txHash). If the updated tx is the readyTx it's returned, nil otherwise
-func (a *addrQueue) UpdateTxZKCounters(txHash common.Hash, counters state.ZKCounters, constraints batchConstraints, weights batchResourceWeights) (readyTxUpdated *TxTracker) {
+// UpdateTxZKCounters updates the ZKCounters for the given tx (txHash)
+// If the updated tx is the readyTx it returns a copy of the previous readyTx, nil otherwise
+func (a *addrQueue) UpdateTxZKCounters(txHash common.Hash, counters state.ZKCounters, constraints batchConstraints, weights batchResourceWeights) (newReadyTx, prevReadyTx *TxTracker) {
 	txHashStr := txHash.String()
 
 	if (a.readyTx != nil) && (a.readyTx.HashStr == txHashStr) {
```

Comment:
I think comparing hashes is faster than comparing strings

Reply:
Yep this is true, because common.Hash values are stored as fixed-size byte arrays, while strings are stored as variable-length arrays of characters. Additionally, comparing two common.Hash values can be done using a simple memory comparison, while comparing two strings requires iterating through each character and comparing them individually. As a result, comparing two common.Hash values is likely to be faster than comparing two strings.